File: NextTaskClient/src/shared/types/invite.ts
// Типы приглашений (централизовано)
// Все комментарии на русском согласно требованиям кодовой базы

export interface WorkspaceRef {
  id: number;
  name: string;
}

export interface UserRef {
  name: string;
  email: string;
}

// Входящее приглашение для текущего пользователя
export interface IncomingInvite {
  id: number;
  token: string;
  workspace: WorkspaceRef;
  inviter: UserRef;
  expires_at?: string;
  status: "pending" | "accepted" | "declined" | "expired";
}

// E-mail приглашение, отправленное из workspace (owner scope)
export interface EmailInvite {
  id: number;
  email: string;
  role?: "owner" | "editor" | "reader";
  created_at: string;
  expires_at?: string;
  status: "pending" | "revoked" | "expired" | "accepted";
}

// Ссылка-приглашение
export interface InviteLinkItem {
  invite_token: string;
  expires_at?: string;
  times_used: number;
  max_uses?: number;
}

File: NextTaskServer/app/schemas/invite.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class WorkspaceRef(BaseModel):
    id: int
    name: str

class UserRef(BaseModel):
    name: str
    email: str

class IncomingInvite(BaseModel):
    id: int
    token: str
    workspace: WorkspaceRef
    inviter: UserRef
    expires_at: Optional[datetime] = None
    status: str  # "pending" | "accepted" | "declined" | "expired"

class IncomingInvitesResponse(BaseModel):
    invites: List[IncomingInvite]

class EmailInvite(BaseModel):
    id: int
    email: str
    role: Optional[str] = None  # "owner" | "editor" | "reader"
    created_at: datetime
    expires_at: Optional[datetime] = None
    status: str  # "pending" | "revoked" | "expired" | "accepted"

class EmailInvitesResponse(BaseModel):
    invites: List[EmailInvite]

class InviteLinkItem(BaseModel):
    invite_token: Optional[str] = None
    token: str
    expires_at: Optional[datetime] = None
    times_used: int
    max_uses: Optional[int] = None

class InviteLinksResponse(BaseModel):
    links: List[InviteLinkItem]

# Legacy snake_case schema for frontend compatibility in invites list
class InviteLinkItemLegacy(BaseModel):
    invite_token: str
    expires_at: Optional[datetime] = None
    times_used: int
    max_uses: Optional[int] = None

class AcceptInviteRequest(BaseModel):
    token: str

class DeclineInviteRequest(BaseModel):
    token: str

class EmailInviteRequest(BaseModel):
    email: str
    role: Optional[str] = None  # "owner" | "editor" | "reader"

class InviteLinkRequest(BaseModel):
    max_uses: Optional[int] = None
    expires_hours: Optional[int] = None
